### PR TITLE
DOCS-2291: Add Optimal Block Size Recommendation

### PR DESCRIPTION
**Pull request by jervin777**
Added optimal block size table and applicable info to go along with it per Joe Maloney's document.

Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

File: content/CORE/Storage/Pools/Zvols.md [hunk 1]
```diff
@@ -36,6 +36,30 @@ To quickly create a Zvol with the default options, enter a name for the Zvol, a
...
+|5/6|Raidz-2|32k|
+|6/7|Raidz-3|32k|
+|6/7/8/9|Raidz-1|64k|
+|7/8/9/10|Raidz-2|64k|
+|8/9/10/11|Raidz-3|64k|
+|10+|Raidz-1|128k|
+|11+|Raidz-2|128k|
+|12+|Raidz-3|128k|
+
+Additional tuning may be required for optimal performance depending on the workload.  See the [workload tuning chapter](https://openzfs.github.io/openzfs-docs/Performance%20and%20Tuning/Workload%20Tuning.html) of the OpenZFS handbook for more information
```
Comment (Mrt134) [inline content/CORE/Storage/Pools/Zvols.md:59]: Discussed with Joe about adding a note about Enterprise support here.
```suggestion
Additional tuning can be required for optimal performance, depending on the workload. iXsystems Engineers are available to assist [Enterprise](https://www.truenas.com/truenas-enterprise/) customers with tuning their TrueNAS hardware. The [workload tuning chapter](https://openzfs.github.io/openzfs-docs/Performance%20and%20Tuning/Workload%20Tuning.html) of the OpenZFS handbook is also a good resource.
```

File: content/CORE/Storage/Pools/Zvols.md [hunk 2]
```diff
@@ -36,6 +36,30 @@ To quickly create a Zvol with the default options, enter a name for the Zvol, a
 
 {{< /expand >}}
 
+{{< expand "Optimal Zvol Block Sizes" "v" >}}
+
+When creating zvols TrueNAS will automatically recommend a sane volume *block size* for optimizing space efficiency.  Below are the minimum volume *block size* values the TrueNAS zvol creation process will recommend. To manually change this value select from the *Block size* dropdown menu.
```
Comment (Mrt134) [inline content/CORE/Storage/Pools/Zvols.md:41]:
```suggestion
TrueNAS automatically recommends a space-efficient *block size* for new zvols. This table shows the minimum volume *block size* values that are recommended. To manually change this value, use the *Block size* dropdown menu.
```

**Review by Mrt134 — COMMENTED** [review]
Quick review and added an element previously discussed with Joe

**Comment by amotin** [comment]
As I have told already, I think this table would look much better with two first columns swapped and sorted on Configuration.

File: content/CORE/Storage/Pools/Zvols.md [hunk 3]
```diff
@@ -36,6 +36,30 @@ To quickly create a Zvol with the default options, enter a name for the Zvol, a
...
+|5|Raidz-3|16K|
+|4/5|Raidz-1|32k|
+|5/6|Raidz-2|32k|
+|6/7|Raidz-3|32k|
+|6/7/8/9|Raidz-1|64k|
+|7/8/9/10|Raidz-2|64k|
+|8/9/10/11|Raidz-3|64k|
+|10+|Raidz-1|128k|
+|11+|Raidz-2|128k|
+|12+|Raidz-3|128k|
```
Comment (Mrt134) [inline content/CORE/Storage/Pools/Zvols.md:57]:
```suggestion
| Configuration | Number of Drives | Optimal Block Size | 
|---------------|-------|-------|
| Mirror | N/A | 16k |
| Raidz-1 | 3 | 16k |
| Raidz-2 | 4 | 16k |
| Raidz-3 | 5 | 16k |
| Raidz-1 | 4/5 | 32k |
| Raidz-2 | 5/6 | 32k |
| Raidz-3 | 6/7 | 32k |
| Raidz-1 | 6/7/8/9 | 64k |
| Raidz-2 | 7/8/9/10 | 64k |
| Raidz-3 | 8/9/10/11 | 64k |
| Raidz-1 | 10+ | 128k |
| Raidz-2 | 11+ | 128k |
| Raidz-3 | 12+ | 128k |
```

Comment (amotin) [reply content/CORE/Storage/Pools/Zvols.md:57]: And sort by the Configuration first.

Comment (Mrt134) [reply content/CORE/Storage/Pools/Zvols.md:57]: Sorry about that, is that cleaner?

Comment (amotin) [reply content/CORE/Storage/Pools/Zvols.md:57]: Sure.

**Review by amotin — APPROVED** [review]

**Review by BonnieF — APPROVED** [review]

**Pull request merged**